frontend/app/.server/routes/helpers/renew-adult-route-helpers.ts

```diff
@@ -0,0 +1,176 @@
+import { getPathById } from '~/utils/route-utils';
+
+interface LoadRenewAdultStateArgs {
+  params: Params;
+  request: Request;
+  session: Session;
+}
+
+/**
+ * Loads renew Adult state.
+ * @param args - The arguments.
+ * @returns The loaded Adult state.
+ */
+export function loadRenewAdultState({ params, request, session }: LoadRenewAdultStateArgs) {
+  const log = getLogger('renew-adult-route-helpers.server/loadRenewAdultState');
+  const { pathname } = new URL(request.url);
+  const renewState = loadRenewState({ params, session });
+
+  if (renewState.typeOfRenewal !== 'adult') {
+    throw redirect(getPathById('public/renew/$id/type-renewal', params));
+  }
+
+  // Redirect to the confirmation page if the application has been submitted and
+  // the current route is not the confirmation page.
+  const confirmationRouteUrl = getPathById('public/renew/$id/adult/confirmation', params);
+  if (renewState.submissionInfo && !pathname.endsWith(confirmationRouteUrl)) {
+    log.warn('Redirecting user to "%s" since the application has been submitted; sessionId: [%s], ', renewState.id, confirmationRouteUrl);
+    throw redirect(confirmationRouteUrl);
+  }
+
+  // Redirect to the first flow page if the application has not been submitted and
+  // the current route is the confirmation page.
+  const termsAndConditionsRouteUrl = getPathById('public/renew/$id/terms-and-conditions', params);
+  if (!renewState.submissionInfo && pathname.endsWith(confirmationRouteUrl)) {
+    log.warn('Redirecting user to "%s" since the application has not been submitted; sessionId: [%s], ', renewState.id, termsAndConditionsRouteUrl);
+    throw redirect(termsAndConditionsRouteUrl);
+  }
+
+  return renewState;
+}
+
+interface LoadRenewAdultStateForReviewArgs {
+  params: Params;
+  request: Request;
+  session: Session;
+}
+
+/**
+ * Loads the renewal state for the review page. It validates the state and throws a redirect if invalid.
+ * If a redirect exception is thrown, state.editMode is set to false.
+ * @param args - The arguments.
+ * @returns The validated adult state.
+ */
+export function loadRenewAdultStateForReview({ params, request, session }: LoadRenewAdultStateForReviewArgs) {
+  const state = loadRenewAdultState({ params, request, session });
+
+  try {
+    return validateRenewAdultStateForReview({ params, state });
+  } catch (err) {
+    if (isResponse(err) && isRedirectResponse(err)) {
+      saveRenewState({ params, session, state: { editMode: false } });
+    }
+    throw err;
+  }
+}
+
+interface ValidateRenewAdultStateForReviewArgs {
+  params: Params;
+  state: RenewState;
+}
+
+export function validateRenewAdultStateForReview({ params, state }: ValidateRenewAdultStateForReviewArgs) {
+  const {
+    applicationYear,
+    hasMaritalStatusChanged,
+    hasAddressChanged,
+    maritalStatus,
+    partnerInformation,
+    contactInformation,
+    editMode,
+    id,
+    submissionInfo,
+    typeOfRenewal,
+    addressInformation,
```

Comment:
I'm currently working on removing `addressInformation` from the parent state object because we now keep `homeAddress` and `mailingAddress` in separate state objects (`addressInformation` is no longer used). Would it be possible to remove any references to `addressInformation` in this route helper and replace with `home/mailingAddress`?